fetch.py

```diff
@@ -0,0 +1,152 @@
+import os
+import time
+import datetime
+import functools
+import zlib
+import requests
+import re
+from tqdm import tqdm
+"""
+Decorator for downloading or updating files required for a function.
+"""
+def reqfile(func=None, path=None, url=None):
+    if not func:
+        return functools.partial(fetch_factory, path=path, url=url)
+    @functools.wraps(test_func)
+    def wrapper(*args, **kwargs):
+        fetch(path, url)
+        return func(*args, **kwargs)
+    return wrapper
+
+"""
+Downloads a file from url to a given path.
+Also shows a progress bar to keep track.
+"""
```

Comment:
Is this docstring for a specific function? If so, put it just below the `def` statement, i.e.

```
def downfile(path, url):
    """
    Downloads a file from url to a given path. 
    Also shows a progress bar to keep track.
    """
```